File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestListOptUUIDValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.test.TypeTester;
import com.dslplatform.ocd.values.ListOptUUIDInValue.ListOptUUIDValue;
import com.dslplatform.patterns.ServiceLocator;
import java.util.*;
import org.junit.*;
import static org.junit.Assert.*;

public class TestListOptUUIDValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
        locator = Bootstrap.init(TestListOptUUIDValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertTrue(TypeTester.testField(ListOptUUIDValue.class, "listOptUUID")
                .resultEquals(List.class, UUID.class));
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertTrue(TypeTester.testGetter(ListOptUUIDValue.class, "getListOptUUID")
                .resultEquals(List.class, UUID.class));
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertTrue(TypeTester.testSetter(ListOptUUIDValue.class, "setListOptUUID", List.class, UUID.class)
                .resultEquals(ListOptUUIDValue.class));
    }
}

File: workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestListOptUUIDValue.scala
package com.dslplatform.ocd.values

import ListOptUUIDInValue.ListOptUUIDValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestListOptUUIDValue extends SpecExtensions {
  def `Test property field type: IndexedSeq[Option[java.util.UUID]]`() =
    checkType(
      typeOf[IndexedSeq[Option[java.util.UUID]]]
    , typeOf[ListOptUUIDValue].member("listOptUUID": TermName).asMethod.returnType
    )

  def `Test property default value: IndexedSeq.empty[Option[java.util.UUID]]`() {
    assert(ListOptUUIDValue().listOptUUID === IndexedSeq.empty[Option[java.util.UUID]])
  }
}
